gitlab/.gitlab-ci-without-askui-studio.yml

```
@@ -1,51 +1,87 @@
-stages:          # List of stages for jobs and their order of execution
+stages:          
   - build
   - test
   - deploy
 
-services:
-  - name: askuigmbh/askui-ui-controller:0.11.2-chrome-129.0-amd64
-    command: ["--port","6769"]
-    alias: uicontroller
-
 default:
   image: node:latest
-  cache:  # Cache modules in between jobs
+  cache:
     key: $CI_COMMIT_REF_SLUG
     paths:
       - .npm/
-  before_script:
-    - npm ci --cache .npm --prefer-offline || true
 
 build-job:
   stage: build
   image: node
   script:
     - npm install
+  artifacts:
+    paths:
+      - node_modules
   only:
     - main
 
 test-job:
   stage: test
+  services:
+    - name: askuigmbh/askui-ui-controller:0.11.2-chrome-129.0-amd64
+      alias: uicontroller
+  before_script:
+    - npm ci --cache .npm --prefer-offline || true
+    # Create base directory
+    - mkdir -p "/root/.askui/SnippingTool/AIElement/${ASKUI_WORKSPACE_ID}"
+    # Create JSON file
+    - |
+      echo '{
+        "version": 1,
+        "id": "{31444F59-104F-4431-9288-42047F712A11}",
+        "creationDateTime": "2024-12-17T12:27:23.763714Z",
+        "image": {
+          "size": {
+            "width": 113,
+            "height": 88
+          }
+        },
+        "name": "cart1",
+        "workspaceId": "'${ASKUI_WORKSPACE_ID}'"
+      }' > "/root/.askui/SnippingTool/AIElement/${ASKUI_WORKSPACE_ID}/{31444F59-104F-4431-9288-42047F712A11}.json"
+    # Create .env file
+    - |
+      cat << EOF > .env
+      standardUserName=$standardUserName
+      commonPassword=$commonPassword
+      lockedUserName=$lockedUserName
+      errorUser=$errorUser
+      problemUser=$problemUser
+      glitchUser=$glitchUser
+      EOF
   script:
-    - npm askui || true
+    - npm run askui || true 
+    - npm install jest @types/jest ts-jest --save-dev  
+    - npm install mocha @types/mocha --save-dev        
```

You're already installing it in line 17.

```suggestion
    - npm run askui || true  
```

---

I think i remember only npm install didnt work, i had to give specific jest and mocha too, but for now ill remove them. maybe there could have been another reason here